langroid/agent/special/doc_chat_agent.py

```diff
@@ -860,6 +878,26 @@ def llm_hypothetical_answer(self, query: str) -> str:
                 ).content
         return answer
 
+    def llm_hypothetical_question(self, chunk: Document) -> Document:
+        """Generate potential questions this chunk would answer"""
+        if self.llm is None:
+            raise ValueError("LLM not set")
+
+        with status("[cyan]LLM generating hypothetical questions..."):
+            with StreamingIfAllowed(self.llm, False):
+                prompt = self.config.hypothetical_questions_default_prompt % {
+                    "passage": chunk.content
+                }
+            questions = self.llm_response_forget(prompt).content
+        return Document(
+            content=questions,
```

Comment:
The content is just the questions, which is fine since it helps match the query, but we lose the original content. I know you are keeping it in the metadata, but `original_content` is not used in the normal flow of this agent anywhere. 

I think it's better to _augment_ the `content` field value with the `questions`, and also demarcate those questions in some special way, so that during passage retrieval, the questions are ignored. I.e. only use the questions for matching/ranking, but ignore in retrieval.

Reply:
The other advantage of augmenting (i.e. combining chunk content + questions) is that now we can match with both the question and content, rather than having to deal with de-duplication, e.g. when we find a match for questions_5 that links to chunk_3 and also find a match to chunk_3.

Reply:
Thanks for the feedback! 

I've implemented the suggested improvements like this:

1. Combined content+questions in a single field instead of separating them, using a consistent marker (`GENERATED_QUESTIONS_MARKER`) to demarcate them.

2. Added `clean_generated_content()` method in the so that questions are used in ranking but not showed in retrieval

Reply:
Thank you @dfm88 ! Added to my queue of things to look at this week.